src/gsy_e/models/area/event_dispatcher.py

```diff
@@ -108,17 +114,25 @@ def broadcast_balancing_market_cycle(self, **kwargs) -> None:
     def _broadcast_notification_to_single_agent(
             self, agent_area: "Area", market_type: AvailableMarketTypes,
             event_type: AreaEvent, **kwargs) -> None:
-        agent_dict = self._get_agents_for_market_type(agent_area.dispatcher, market_type)
-        for time_slot, agent in agent_dict.items():
-            if time_slot not in agent_area.get_market_instances_from_class_type(
-                    market_type):
-                # exclude past IAAs
-                continue
 
-            agent.event_listener(event_type, **kwargs)
+        if market_type == AvailableMarketTypes.FUTURE:
+            # First check for the market_type and then check if an future agent exists, otherwise
+            # the future event would be posted on non-future agents.
+            if self.future_agent:
+                self.future_agent.event_listener(event_type, **kwargs)
```

Comment:
```suggestion
        if market_type == AvailableMarketTypes.FUTURE and self.future_agent:
            # First check for the market_type and then check if an future agent exists, otherwise
            # the future event would be posted on non-future agents.
            self.future_agent.event_listener(event_type, **kwargs)
```

Reply:
@Hassan754 please read carefully, I explicitly left an inline comment in response to a comment from Faizan. This needs to be like I implemented.

Reply:
@hannesdiedrich I'm not sure I get you, how is 
```python
if condition_a and condition_b:
    do_something()
```
different than

```python
if condition_a
   if condition_b:
     do_something()
```

Reply:
The problem is the else block
```python
if condition_a and condition_b:
    do_something()
else:
    do_something_else()
```
is different from 
```python
if condition_a
   if condition_b:
    do_something()
else:
    do_something_else()
```
In the first case, if one of the conditions is not try, it will automatically go into the else block. But this is not what we want. we actually want
```python
if condition_a
   if condition_b:
     do_something()
   else:
      do_nothing()
else:
    do_something_else()
```
I you like i can add the specific else block or convert the outer else block into an and check explicitly to for the rest of the market types.

Reply:
Ah I see, I completely overlooked this
In this case, you have the freedom to select any approach you like

Reply:
My proposal:
```python
if condition_a and condition_b:
       do_something()
if condition_c:
    do_something_else()
```

Reply:
Seems great! only convert the second if to a `elif` and it should be fine

Reply:
this will make pylint complain. :-( 
I will go the following way actually:
```python
if condition_a and condition_b:
       do_something()
       return
do_something_else()
```

Reply:
I tried this and pylint didn't complain
```python
        if market_type == AvailableMarketTypes.FUTURE and self.future_agent:
        elif market_type != AvailableMarketTypes.FUTURE:
```